contributed_definitions/nyaml/NXpeak.yaml

```diff
@@ -1,59 +1,140 @@
 category: base
 doc: |
-  Description of peaks, their functional form or measured support.
+  Base class for describing spectral peaks, their functional form, and support values
+  (i.e., the discretization (points) at which the function has been evaluated).
 symbols:
   doc: |
     The symbols used in the schema to specify e.g. dimensions of arrays.
   n_support: |
-    Number of support points
+    Number of support points.
 type: group
 NXpeak(NXobject):
   label:
     doc: |
-      Human-readable identifier to specify which concept/entity
+      Human-readable label which specifies which concept/entity
       the peak represents/identifies.
-  peak_model:
+  lineshape:
     doc: |
       Is the peak described analytically via a functional form
       or is it empirically defined via measured/reported
-      intensity/counts as a function of an independent variable.
+      intensity/counts as a function of an independent variable?
       
-      If the functional form is not empirical or gaussian, users
-      should enter other for the peak_model and add relevant details
+      If the functional form is not empirical or Gaussians, users
+      should enter other for the description and add relevant details
       in the NXcollection.
-    enumeration: [empirical, gaussian, lorentzian, other]
+    enumeration: [empirical, gaussian, lorentzian, voigt, other]
+  lineshape_other:
+    doc: |
+      Human-readable descritpion of the lineshape used for the peak if it does
+      not fit into the enumeration given in `lineshape`.
   position(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the position values for the independent variable.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
+    dim: (n_support,)
   intensity(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the intensity/count values at each position.
```

Comment:
Is this for the final, fitted result? Or just the functional form? Why not also use this for other elements in the enumeration, only "emperical" (whatever that means for a peak)?

Reply:
This is indeed meant for the final fitted result. The functional form should be described in a different way, as you suggested above.